independent-projects/resteasy-reactive/common/runtime/src/main/java/org/jboss/resteasy/reactive/common/util/MediaTypeHelper.java

```diff
@@ -267,10 +267,12 @@ public static boolean equivalentParams(MediaType m1, MediaType m2) {
         return true;
     }
 
+    // TODO: does this need to be more complex?
     public static boolean isTextLike(MediaType mediaType) {
-        return "text".equalsIgnoreCase(mediaType.getType())
-                || ("application".equalsIgnoreCase(mediaType.getType())
-                        && mediaType.getSubtype().toLowerCase().startsWith("xml"));
+        String type = mediaType.getType();
+        String subtype = mediaType.getSubtype();
+        return (type.equals("application") && (subtype.contains("json") || subtype.contains("xml") || subtype.contains("yaml")))
+                || type.equals("text");
```

Comment:
Heh, I wonder if there's any more formal definition of what can be turned into a text.

Also, I wonder how this all compares to "toplevel" body parsing.

In _theory_, multipart members and body entities should now have the same parsing mechanism, in that they each have an encoding, content type, and go via `MessageBodyReader` for deserialisation, which is what allows us to have not just primitive types but also JSON/XML deserialisation.

This logic appears to sit _before_ we look up the right `MessageBodyReader` for the deserialisation. I wonder why we have this. Perhaps everything _at this point_ should be read as `byte[]` (well, and associated with their content-type), and then later when we figure out whether to deserialise this into a `String` or `JsonObject` or `Person` entity, we use the same binary data?

Here we're accepting "deserialising" XML/JSON/YAML into `String` which means: no deserialisation, just give them "raw" (still respecting the encoding). For a mime part. Do we accept the same for a toplevel http request body? I think we do. But I wonder if we also limit it there for XML/JSON/YAML or if we have another sort of check of text-like, or… "String-convertible".

Reply:
The whole thing is weird and this is only needed to try to decide what type of part of this is (see calls to `FormData#add`).
I really don't like what we are doing, but I don't have any ideas to make it better.